tls/client-tls13-resume.c

```diff
@@ -0,0 +1,291 @@
+    if ((ssl = wolfSSL_new(ctx)) == NULL) {
+        fprintf(stderr, "ERROR: failed to create WOLFSSL object\n");
+        ret = -1;
+        goto exit;
+    }
+
+    /* Attach wolfSSL to the socket */
+    wolfSSL_set_fd(ssl, sockfd);
+
+    /* Connect to wolfSSL on the server side */
+    if ((ret = wolfSSL_connect(ssl)) != WOLFSSL_SUCCESS) {
+        fprintf(stderr, "ERROR: failed to connect to wolfSSL\n");
+        goto exit;
+    }
+
+
+
+    /* Get a message for the server from stdin */
+    printf("Message for server: ");
+    memset(buff, 0, sizeof(buff));
+    if (fgets(buff, sizeof(buff), stdin) == NULL) {
+        fprintf(stderr, "ERROR: failed to to get message for server\n");
+        ret = -1; 
+        goto exit;
+    }
+    len = strnlen(buff, sizeof(buff));
+
+    /* Send the message to the server */
+    if ((ret = wolfSSL_write(ssl, buff, len)) != len) {
+        fprintf(stderr, "ERROR: failed to write\n");
+        goto exit;
+    }
+
+
+
+    /* Read the server data into our buff array */
+    memset(buff, 0, sizeof(buff));
+    if (wolfSSL_read(ssl, buff, sizeof(buff)-1) < 0) {
+        fprintf(stderr, "ERROR: failed to read\n");
+        ret = -1; 
+        goto exit;
+    }
+
+    /* Print to stdout any data the server sends */
+    printf("Server: %s\n", buff);
+
+
+
+    /* Save the session */
+    session = wolfSSL_get_session(ssl);
+
+    /* Close the socket */
+    wolfSSL_free(ssl);
+    close(sockfd);
+
+
+
+    /* --------------------------------------- *
+     * we are now disconnected from the server *
+     * --------------------------------------- */
+
+
+
+    /* Create a new WOLFSSL object to resume with */
+    if ((sslRes = wolfSSL_new(ctx)) == NULL) {
+        fprintf(stderr, "ERROR: failed to create WOLFSSL object\n");
+        ret = -1;
+        goto exit;
+    }
+
+    /* Set up to resume the session */
+    if ((ret = wolfSSL_set_session(sslRes, session)) != WOLFSSL_SUCCESS) {
+        fprintf(stderr, "ERROR: failed to set session\n");
+        goto exit;
+    }
+
+
+
+    /* Get a new socket */
+    if ((sockfd = socket(AF_INET, SOCK_STREAM, 0)) == -1) {
+        fprintf(stderr, "ERROR: failed to create the socket\n");
+        ret = -1; 
+        goto exit;
+    }
+
+
+
+    /* Reconnect to the server */
+    if (connect(sockfd, (struct sockaddr*) &servAddr, sizeof(servAddr))
+        == -1) {
+        fprintf(stderr, "ERROR: failed to connect\n");
+        ret = -1;
+        goto exit;
+    }
+
+    /* Attach wolfSSL to the socket */
+    wolfSSL_set_fd(sslRes, sockfd);
+
+    /* Reconnect to wolfSSL */
+    if ((ret = wolfSSL_connect(sslRes)) != WOLFSSL_SUCCESS) {
+        fprintf(stderr, "ERROR: failed to connect to wolfSSL\n");
+        goto exit;
+    }
+
+
+
+    /* Test if the resume was successful */
+    if (wolfSSL_session_reused(sslRes)) {
+        printf("Session ID reused; Successful resume.\n");
+    }
+    else {
+        printf("Session ID not reused; Successful resume.\n");
```

Comment:
This should say `resume not successful` or something like `Session not resumed, full handshake done instead`.